pkg/server/api/debug/v1/service.go

```diff
@@ -144,13 +104,58 @@ func (s *Service) GetInfo(ctx context.Context, req *debug_pb.GetInfoRequest) (*d
 			EntriesCount:          entries.Entries,
 			FederatedBundlesCount: bundles.Bundles,
 			SvidChain:             svidChain,
-			Uptime:                s.uptime.Unix(),
+			Uptime:                int32(s.uptime().Seconds()),
 		}
 	}
 
 	return s.getInfoResp.resp, nil
 }
 
+func (s *Service) getCertificateChain(ctx context.Context, log logrus.FieldLogger) ([]*debug.GetInfoResponse_Cert, error) {
+	trustDomainID := s.td.IDString()
+
+	// Extract trustdomains bundle and append federated bundles
+	bundle, err := s.ds.FetchBundle(ctx, &datastore.FetchBundleRequest{
+		TrustDomainId: trustDomainID,
+	})
+	if err != nil {
+		return nil, api.MakeErr(log, codes.Internal, "failed to fetch trust domain bundle", err)
+	}
+
+	if bundle.Bundle == nil {
+		return nil, api.MakeErr(log, codes.NotFound, "bundle for trust domain not found", nil)
+	}
+
+	// Create bundle source using rootCAs
+	var rootCAs []*x509.Certificate
+	for _, b := range bundle.Bundle.RootCas {
+		cert, err := x509.ParseCertificate(b.DerBytes)
+		if err != nil {
+			return nil, api.MakeErr(log, codes.Internal, "failed to parse bundle", err)
+		}
+		rootCAs = append(rootCAs, cert)
+	}
+	bundleSource := x509bundle.FromX509Authorities(s.td, rootCAs)
+
+	// Verify certificate to extract SVID chain
+	_, certs, err := x509svid.Verify(s.so.State().SVID, bundleSource)
+	if err != nil {
+		return nil, api.MakeErr(log, codes.Internal, "failed verification against bundle", err)
+	}
+
+	// Create SVID chain for response
+	var svidChain []*debug.GetInfoResponse_Cert
+	for _, root := range certs[0] {
+		svidChain = append(svidChain, &debug_pb.GetInfoResponse_Cert{
+			Id:        spiffeIDFromCert(root),
```

Comment:
There is no requirement that any of the CA certificates (intermediate or otherwise) have SPIFFE IDs. In fact, it is probably more likely in the upstream CA case, that they don't. The ID should be optional.

Reply:
it is optional code is returning nil as ID if no SPIFFE ID found https://github.com/spiffe/spire/pull/1792/files/79c70c5100674ce184fcabdce0d3136563d5e720..b9f1fff48099b243e1e98b4648df0e221165df38#diff-006dc77efc9b2e475c945da4c1b8396fR162